lib/rules/indent.js

```diff
@@ -0,0 +1,832 @@
+         }
+
+         return false;
+      }
+
+      /**
+      * Check indent for array block content or object block content
+      * @param {ASTNode} node node to examine
+      * @returns {void}
+      */
+      function checkIndentInArrayOrObjectBlock(node) {
+         var parentVarNode = getVariableDeclaratorNode(node),
+             nodeIndent, elementsIndent, elements, parent, effectiveParent;
+
+         // Skip inline
+         if (isSingleLineNode(node)) {
+            return;
+         }
+
+         elements = (node.type === 'ArrayExpression') ? node.elements : node.properties;
+
+         // filter out empty elements example would be [ , 2] so remove first element as espree considers it as null
+         elements = elements.filter(function(elem) {
+            return elem !== null;
+         });
+
+         // Skip if first element is in same line with this node
+         if (elements.length > 0 && elements[0].loc.start.line === node.loc.start.line) {
+            return;
+         }
+
+
+         // TODO - come up with a better strategy in future
+         if (isNodeFirstInLine(node)) {
+            parent = node.parent;
+            effectiveParent = parent;
+
+            if (parent.type === 'MemberExpression') {
+               if (isNodeFirstInLine(parent)) {
+                  effectiveParent = parent.parent.parent;
+               } else {
+                  effectiveParent = parent.parent;
+               }
+            }
+            nodeIndent = getNodeIndent(effectiveParent);
+            if (parentVarNode && parentVarNode.loc.start.line !== node.loc.start.line) {
+               if (parent.type !== 'VariableDeclarator' || parentVarNode === parentVarNode.parent.declarations[0]) {
+                  if (parent.type === 'VariableDeclarator' && parentVarNode.loc.start.line === effectiveParent.loc.start.line) {
+                     // eslint-disable-next-line max-len
+                     nodeIndent = nodeIndent + (indentSize * options.VariableDeclarator[parentVarNode.parent.kind]) + options.VariableDeclaratorOffset[parentVarNode.parent.kind];
+                  } else if (
+                     parent.type === 'ObjectExpression' ||
+                     parent.type === 'ArrayExpression' ||
+                     parent.type === 'CallExpression' ||
+                     parent.type === 'ArrowFunctionExpression' ||
+                     parent.type === 'NewExpression'
+                  ) {
+                     nodeIndent = nodeIndent + indentSize;
+                  }
+               }
+            } else if (
+               !parentVarNode && !isFirstArrayElementOnSameLine(parent) &&
+               effectiveParent.type !== 'MemberExpression' &&
+               effectiveParent.type !== 'ExpressionStatement' &&
+               effectiveParent.type !== 'AssignmentExpression' &&
+               effectiveParent.type !== 'Property') {
```

Comment:
Not only does this block break it's own indentation rule and is difficult to read and understand, it's just plain ugly. Could we perhaps refactor the conditions into a variable or two that will fit onto one line in the if statement?

i.e.

```
isConditionOne = !parentVarNode && !isFirstArrayElementOnSameLine(parent);
isConditionTwo = (['MemberExpression', 'ExpressionStatement',  'AssignmentExpression',  'Property'].indexOf(effectiveParent.type) === -1);
...
} else if (isConditionOne && isConditionTwo) {
```

with better variable names, of course :)

I'm also opening another issue in the `eslint-config-silvermine` repo to disallow multi-line `if` statements, so please eliminate those throughout this file.